test/api/calendar.test.ts

```diff
@@ -67,3 +68,59 @@ describe('the getInbox method', () => {
     });
   })
 });
+
+describe('the modifyEvent method', () => {
+  beforeEach(() => {
+    httpClient = {
+      request: jest.fn(),
+      requestJson: jest.fn(),
+      requestText: jest.fn(),
+    };
+
+    davClient = new DAVClient({
+      baseURL: 'http://url',
+      httpClient,
+      headers: {
+        Authorization: 'Basic header',
+      },
+    });
+  });
+
+  it('should send a PUT request to the event path with the event object ICS as body', () => {
+    const path = 'events/123456.ics';
+    const testEventObject: CalendarEventObject = {
+      id: '123456',
+      title: 'test',
+      start: '2021-03-15T10:00:00',
+      end: '2021-03-15T11:00:00',
+      allDay: false,
+      description: 'simple description',
+      location: 'some location',
+      duration: {
+        hours: 1,
+      },
+      extendedProps: {},
+    };
+
+    modifyEvent(davClient)(path, testEventObject);
+    expect(httpClient.requestText).toHaveBeenCalledWith({
+      url: 'http://url/calendars/events/123456.ics',
+      method: 'PUT',
+      body: 'BEGIN:VCALENDAR\r\n\
+VERSION:2.0\r\n\
+CALSCALE:GREGORIAN\r\n\
+BEGIN:VEVENT\r\n\
+UID:123456\r\n\
+SUMMARY:test\r\n\
+LOCATION:some location\r\n\
+DESCRIPTION:simple description\r\n\
+DTSTART:20210315T100000\r\n\
+DTEND:20210315T110000\r\n\
+END:VEVENT\r\n\
+END:VCALENDAR',
```

Comment:
You're also testing the integrity of the `dav-parser` here?

Reply:
So this is more like an integration test. Is this what you intended to do?

Reply:
nope, the translator is tested in the `dav-parser` library with a variation of calendar event objects.
I'm here simply testing the parameters that the `requestText` was called with.

Reply:
Then you should mock the `translate` function of `dav-parser` here.

Reply:
The true unit test requires the mocking of `url-join` also :joy:.

Reply:
But hold on did you already do the same for the tests for `getInbox`? If so... proceed :joy:.

Reply:
no need to do anything related to `url-join` in the tests, it works out of the box